internal/engine/image_build_and_deployer.go

```diff
@@ -346,10 +355,12 @@ func (ibd *ImageBuildAndDeployer) createEntitiesToDeploy(ctx context.Context,
 					var sidecarInjected bool
 					e, sidecarInjected, err = sidecar.InjectSyncletSidecar(e, injectedRefSelector, ibd.syncletContainer)
 					if err != nil {
-						return nil, err
+						return result, err
 					}
 					if !sidecarInjected {
-						return nil, fmt.Errorf("Could not inject synclet: %v", e)
+						result.syncletInjectionError = true
```

Comment:
waiiiitasec, this name is confusing: this is the case where it didn't throw an ERROR, we just failed to inject it when we expected to. can you rename to clarify? maybe `failedToInject`?

Reply:
yep, it's now called injectedImageWithoutSynclet